### ISSUE TITLE
OpenSSL 1.1.1b security warning on Google Play

### ISSUE THREAD

**Issue by StudentGGG** [chaquo/chaquopy#727]
On my last app update I got a security warning from Google play:

`Defective OpenSSL version: Your app uses a defective version of the OpenSSL library, which can cause your app to crash. Update to a fixed version of OpenSSL. Defective versions: OpenSSL 1.1.1b`

I suspect it comes from chaquopy because using this trick I could track down that most usage of OpenSSL comes from Chaquopy
https://stackoverflow.com/questions/24197777/google-play-and-openssl-warning-message

Apparently there is a fix in OpenSSL 1.1.1i or higher:
https://support.google.com/faqs/answer/12576638

I am using Chaquopy 12.0.
Could you confirm whether Chaquopy 12.0 uses OpenSSL 1.1.1.b or not?
If yes, is there a plan to publish a fix ?

**Comment by mhsmith** [comment]
Yes, Chaquopy does use this version of OpenSSL. We'll shortly be releasing a new Chaquopy version which adds support for Python 3.9, 3.10 and 3.11. All of these will use OpenSSL 3, which will resolve this issue.

We're not planning to update the OpenSSL version of Python 3.8 in the next Chaquopy release, but we'll do it in the release after that.

**Comment by mhsmith** [comment]
Here's how to make your own build to remove the Google Play warning:

Use a Linux machine: a cheap virtual server from somewhere like DigitalOcean will do just fine.

Update Common.java to set the correct build number: 1 for Chaquopy 12.0, 2 for Chaquopy 13.0:
https://github.com/chaquo/chaquopy/blob/18141605d0b4713230c06d912be666ac87e3fd43/product/buildSrc/src/main/java/com/chaquo/python/Common.java#L15-L16

Install Docker using the [instructions on its website](https://docs.docker.com/install/#supported-platforms), then:
```
cd target
docker build -t chaquopy-target .
```

Download the Android SDK command-line tools from [here](https://developer.android.com/studio#command-tools), and set them up as follows (replace `PATH/TO` with the correct location):
```
mkdir -p android-sdk/cmdline-tools
cd android-sdk/cmdline-tools
unzip PATH/TO/commandlinetools-linux-8512546_latest.zip
mv cmdline-tools latest
```

Find this section of target/build-all.sh:
https://github.com/chaquo/chaquopy/blob/18141605d0b4713230c06d912be666ac87e3fd43/target/build-all.sh#L25-L26

And edit it as follows:
* Change the OpenSSL version to something newer, e.g. `1.1.1r`.
* Comment out the bottom section of the file which builds OpenSSL 3 and Python 3.9-3.11.

Run the build (replace `PATH/TO` with the correct location):
```
cd target
docker run -it -v PATH/TO/android-sdk:/root/android-sdk -v $(pwd):/root/target -v $(pwd)/../product:/root/product  -v $(pwd)/../maven:/root/maven chaquopy-target
```

You should now have some Python 3.8 builds in maven/com/chaquo/python/target. To use them:
* Find the `pluginManagement { repositories }` section of your app's settings.gradle file, or the `buildscript { repositories }` section of your `build.gradle` file.
* At the *start* of the `repositories` section, add the following line (replace `PATH/TO` with the correct location):
  ```
  maven { url "PATH/TO/chaquopy/maven" }
  ```
* Rebuild your app.

**Comment by mhsmith** [comment]
This issue was fixed in Chaquopy 14.0.2. To upgrade, edit your app's top-level build.gradle file and change the version number of `com.chaquo.python`.

**Comment by mhsmith** [comment]
This warning may also be produced by older versions of the `cryptography` package, which were linked statically against OpenSSL 1.1 on all Python versions. To fix this:

* Set your [app's Python version](https://chaquo.com/chaquopy/doc/current/android.html#python-version) to 3.10 or later.
* Set your `cryptography` version to 42.0.8 or later.
